File: problem/bool-not/6/main.c
/*
 * main.c: main file for solutions to the "bool-not" exercise
 *
 * This is to be linked to the submitted file.
 * It processes values from standard input then from the "in.txt" file.
 *
 *
 * Copyright (C) 2020  Rudy Matela
 * All rights reserved.
 */
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
int negation(int);
void process(FILE *fi)
{
	char p[6];
	int b;
    while (fscanf(fi, " %s", p)==1) {
        b = strcmp(p, "true") == 0;
        b = negation(b);
        puts(b ? "true" : "false");
    }
}
void errxit(const char *msg)
{
	fprintf(stderr, msg);
	exit(1);
}
int main()
{
	int i, j;
	FILE *f;
	process(stdin);
	f = fopen("in.txt", "r");
	if (!f) errxit("could not open in.txt");
	process(f);
	fclose(f);
	return 0;
}

File: problem/bool-not/6/main.hs
-- main.hs: main file for solutions to the "bool-not" exercise
--
-- Copyright (C) 2020  Rudy Matela
-- All rights reserved.
--
--
-- This file is part of the CSCX Online Judge.
--
-- This is to be linked to the submitted file.
-- It processes values from standard input then from the "in.txt" file.
module AltMain (main) where

import Main (negation)
import Data.Char (toLower)

readBool :: String -> Bool
readBool "true"   =  True
readBool "false"  =  False
readBool s        =  read s

io :: String -> String
io = unlines . map (map toLower . show . negation . readBool) . lines

main :: IO ()
main = do
  -- processes standard input
  interact io

  -- processes in.txt
  more <- readFile "in.txt"
  putStr $ io more
